### PR TITLE
sessionctx/binloginfo: migrate test-infra to testify for sessionctx/binloginfo (#27279)

### LINKED ISSUE #27279: migrate test-infra to testify for `sessionctx/binloginfo`

### PR DESCRIPTION
**Pull request by Defined2014**
### What problem does this PR solve?

Issue Number: close #27279 

### Release note 

```release-note
None
```

### CODE REVIEW

**Comment by Defined2014** [comment]
/cc @tisonkun PTAL
For easy to implement, I used stretchr/testify/suite to instead of orignal suite. Also changed testkit/testkit.go and tools/check/check_testSuite.sh. The reason is below:
* testkit.go: follow the util/testkit/testkit.go logical, create session when call Exec. Because session.BootstrapSession() func will close the session pool.
* check_testSuite.sh: I still use suite, but didn't call "Suite|SerialSuites" func. I take a work around to fix it by change the shell script. I think you can rewrite the shell script when migrate finish.

**Review by tisonkun — CHANGES_REQUESTED** [review]
as mention in the tracking issue:

> Avoid using suite of testify as discussed https://github.com/pingcap/tidb/pull/26082#issuecomment-877230972

You can refer to other test case where `testkit.CreateMockStore` in use. And try to batch setup / teardown logic only if necessary.

File: sessionctx/binloginfo/binloginfo_test.go [hunk 1]
```diff
@@ -293,37 +289,37 @@ func (s *testBinlogSuite) TestMaxRecvSize(c *C) {
 	}
 	binlogWR := info.WriteBinlog(1)
 	err := binlogWR.GetError()
-	c.Assert(err, NotNil)
-	c.Assert(terror.ErrCritical.Equal(err), IsFalse, Commentf("%v", err))
+	require.NotNil(t, err)
```
Comment (tisonkun) [inline sessionctx/binloginfo/binloginfo_test.go:292]: `require.Error(t, err)` ditto others

File: sessionctx/binloginfo/binloginfo_test.go [hunk 2]
```diff
@@ -293,37 +289,37 @@ func (s *testBinlogSuite) TestMaxRecvSize(c *C) {
...
 			break
 		}
 	}
 	pump.mu.Unlock()
-	c.Assert(bin, NotNil)
+	require.NotNil(t, bin)
 	preVal := new(binlog.PrewriteValue)
 	err := preVal.Unmarshal(bin.PrewriteValue)
-	c.Assert(err, IsNil)
+	require.Nil(t, err)
```
Comment (tisonkun) [inline sessionctx/binloginfo/binloginfo_test.go:312]: `require.NoError(t, err)` ditto others.

File: testkit/testkit.go [hunk 3]
```diff
@@ -169,10 +174,3 @@ func (tk *TestKit) ExecToErr(sql string, args ...interface{}) error {
 	}
 	return err
 }
-
-func newSession(t *testing.T, store kv.Storage) session.Session {
```
Comment (tisonkun) [inline testkit/testkit.go:173]: why make this change?

**Review by zimulala — APPROVED** [review]
LGTM

**Pull request merged**